README.md

```diff
@@ -219,91 +210,138 @@ import java.sql.*;
+
+If custom connection plugins are loaded, Enhanced Failure Monitoring will NOT be loaded unless explicitly included by adding `com.mysql.cj.jdbc.ha.ca.plugins.NodeMonitoringConnectionPluginFactory` to `connectionPluginFactories`. 
+
+#### Enhanced Failure Monitoring Parameters
+`failureDetectionTime`, `failureDetectionInterval`, and `failureDetectionCount` are similar to TCP Keep Alive parameters.
+
+Additional monitoring configurations can be included by adding the prefix `monitoring-` to the configuration key.
+
+| Parameter       | Value           | Required      | Description  | Default Value |
+| ------------- |:-------------:|:-------------:|:-------------:| --------- |
+|`failureDetectionEnabled` | Boolean | No | Set to false to disable enhanced failure monitoring feature. | `true` |
+|`failureDetectionTime` | Integer | No | Interval in milliseconds between sending a SQL query to the server and the first probe to the database node. | `30000` |
+|`failureDetectionInterval` | Integer | No | Interval in milliseconds between probes to database node. | `5000` |
+|`failureDetectionCount` | Integer | No | Number of failed connection checks before considering database node as unhealthy. | `3` |
+|`monitorDisposalTime` | Integer | No | Interval in milliseconds for a monitor to be considered inactive and to be disposed. | `60000` |
+
 ## Extra Additions
 
 ### XML Entity Injection Fix
 
 The default XML parser contained a security risk which made the driver prone to XXE (XML Entity Injection) attacks. To solve this issue, we disabled DTDs (Document type definition) in the XML parser. If you require this restriction to be lifted in your application, you can set the `allowXmlUnsafeExternalEntity` parameter in the connection string to true. Please see table below for a definition of this parameter. 
 
-| Parameter       | Value           | Required      | Description  |
-| ------------- |:-------------:|:-------------:| ----- |
-|`allowXmlUnsafeExternalEntity` | Boolean | No | Set to true if you would like to use XML inputs that refer to external entities. WARNING: Setting this to true is unsafe since your system to be prone to XXE attacks.<br/><br/>**Default value:** `false` |
+| Parameter       | Value           | Required      | Description  | Default Value |
+| ------------- |:-------------:|:-------------:|:-------------:| --------- |
+|`allowXmlUnsafeExternalEntity` | Boolean | No | Set to true if you would like to use XML inputs that refer to external entities. WARNING: Setting this to true is unsafe since your system to be prone to XXE attacks. | `false` |
 
 ### AWS IAM Database Authentication
 
-An optional authentication method is to use Amazon AWS Identity and Access Management (IAM). 
+An optional authentication method is to use Amazon AWS Identity and Access Management (IAM). Use of IAM requires RDS SDK to be added to the classpath.
```

Comment:
https://github.com/awslabs/aws-mysql-jdbc#aws-iam-database-authentication
I like this version more. Maybe we should leave it untouched or update main and dev with the latest changes from upstream.